src/Data/Coyoneda.purs

```diff
@@ -78,6 +102,21 @@ unCoyoneda :: forall f g a. (forall b. (b -> a) -> f b -> g a) -> Coyoneda f a -
 unCoyoneda f (Coyoneda e) = runExists (\(CoyonedaF k fi) -> f k fi) e
 
 -- | Lift a value described by the type constructor `f` to `Coyoneda f`.
+-- |
+-- | Note that for any `f` `liftCoyoneda` has a right inverse
```

Comment:
Good catch. For any f it has a left inverse.

ps. I will have more pr like this one.